src/main/java/com/google/cloud/genomics/dataflow/readers/ReadStreamer.java

```diff
@@ -0,0 +1,144 @@
+            "readGroupSets/id,nextPageToken");
+    for (ReadGroupSet r : rgs) {
+      output.add(r.getId());
+    }
+    if (output.isEmpty()) {
+      throw new IOException("Dataset " + datasetId + " does not contain any ReadGroupSets");
+    }
+    return output;
+  }
+
+  /**
+   * Constructs a StreamReadsRequest for a readGroupSetId, assuming that the user wants all
+   * to include all references.
+   */
+  public static StreamReadsRequest getReadRequests(String readGroupSetId) {
+    return StreamReadsRequest.newBuilder()
+        .setReadGroupSetId(readGroupSetId)
+        .build();
+  }
+
+  /**
+   * Constructs a StreamReadsRequest for a readGroupSetId and a set of user provided references.
+   */
+  public static List<StreamReadsRequest> getReadRequests(final String readGroupSetId,
+      String references) {
+    final Iterable<Contig> contigs = Contig.parseContigsFromCommandLine(references);
+    return FluentIterable.from(contigs)
+        .transformAndConcat(new Function<Contig, Iterable<Contig>>() {
+            @Override
+            public Iterable<Contig> apply(Contig contig) {
+              return contig.getShards(100000000L);
+            }
+          })
+        .transform(new Function<Contig, StreamReadsRequest>() {
+          @Override
+          public StreamReadsRequest apply(Contig shard) {
+            return StreamReadsRequest.newBuilder()
+            .setReadGroupSetId(readGroupSetId)
+            .setStart(shard.start)
+            .setEnd(shard.end)
+            .setReferenceName(shard.referenceName)
+            .build();
+          }
+        }).toList();
+
+  }
+
+  /**
+   * PTransform for streaming reads via gRPC.
+   */
+  public static class StreamReads extends
+      PTransform<PCollection<StreamReadsRequest>, PCollection<Read>> {
+
+    @Override
+    public PCollection<Read> apply(PCollection<StreamReadsRequest> input) {
+      return input.apply(ParDo.of(new RetrieveReads()))
+          .apply(ParDo.of(new ConvergeReadsList()));
+    }
+  }
+
+  private static class RetrieveReads extends DoFn<StreamReadsRequest, List<Read>> {
+
+    private transient StreamingReadServiceGrpc.StreamingReadServiceBlockingStub readStub;
+
+    @Override
+    public void startBundle(Context c) throws IOException {
+      this.readStub = StreamingReadServiceGrpc.newBlockingStub(Channels.fromDefaultCreds());
+    }
+
+    @Override
+    public void processElement(ProcessContext c) {
+      Iterator<StreamReadsResponse> iter = readStub.streamReads(c.element());
+      while (iter.hasNext()) {
+        StreamReadsResponse readResponse = iter.next();
+        c.output(readResponse.getAlignmentsList());
+      }
+    }
+  }
+
+  /**
+   * This step exists to emit the individual reads in a parallel step to the StreamReads step in
+   * order to increase throughput.
+   */
+  private static class ConvergeReadsList extends DoFn<List<Read>, Read> {
```

Comment:
tiny nit (and please feel free to ignore): the function name `ConvergeReadsList` did not convey the function's purpose to me at first glance.  I looked up the dictionary definition for "converge" and perhaps it is appropriate in this case because the scenario is many lists coming together as a collection of individual reads?  

Alternative names could be ConvertReadsListToReads or EmitIndividualReads.